src/components/ResultsBody/PaginatedList/index.jsx

```diff
@@ -0,0 +1,54 @@
+import styles from "./paginatedList.module.css";
+import React, { useState, useEffect } from "react";
+import { Pagination, Box, Card, Typography } from "@mui/material";
+import { useAllVenues } from "./usePaginatedList.js";
+import { VenueCard } from "./VenueCard/index.jsx";
+
+export const PaginatedList = ({ numberOfCards }) => {
+  const itemsPerPage = numberOfCards;
+  const [page, setPage] = useState(1);
+
+  const { venues, isLoading } = useAllVenues();
+
+  if (isLoading) {
+    return <Typography>Loading...</Typography>;
+  }
+
+  const startIndex = (page - 1) * itemsPerPage;
+  const endIndex = startIndex + itemsPerPage;
+
+  const displayedItems = venues ? venues.slice(startIndex, endIndex) : [];
```

Comment:
`.slice` iterates underneath. You are fetching all the venues, in such a case you should use `useMemo` hook. 

With the proper pagination like mentioned in the comment under, this case would not happen, because you would have max 36 venues which is O(1) complexity.

Reply:
done